### PR TITLE
Improve Python docstring for operations

### PR DESCRIPTION
**Pull request by pepone**
This PR improves the generated doc string for Python operations, the same code is used for both proxy and servant operations.

### CODE REVIEW

File: cpp/src/slice2py/PythonUtil.cpp [hunk 1]
```diff
@@ -200,8 +292,6 @@ namespace Slice
             {
                 DocSync,
                 DocAsync,
-                DocAsyncBegin,
-                DocAsyncEnd,
```
Comment (pepone) [inline cpp/src/slice2py/PythonUtil.cpp:204]: Those were no longer used, they correspond to the old bening/end API.

**Review by externl — APPROVED** [review]

File: cpp/src/slice2py/PythonUtil.cpp [hunk 2]
```diff
@@ -2520,122 +2604,139 @@ Slice::Python::CodeVisitor::writeDocstring(const OperationPtr& op, DocstringMode
...
-            _out << nl << contextParamName << " -- The request context for the invocation.";
+            _out << nl << contextParamName << " : Ice.Context";
+            _out << nl << "    The request context for the invocation.";
         }
         if (mode == DocDispatch || mode == DocAsyncDispatch)
         {
             const string currentParamName = getEscapedParamName(op, "current");
-            _out << nl << currentParamName << " -- The Current object for the invocation.";
+            _out << nl << currentParamName << " : Ice.Current";
+            _out << nl << "    The Current object for the invocation.";
```
Comment (bernardnormier) [inline cpp/src/slice2py/PythonUtil.cpp:2648]: We should not mix-up invocation and dispatch.
client-side including client-side context => invocation
server-side including server-side current => dispatch

File: cpp/src/slice2py/PythonUtil.cpp [hunk 3]
```diff
@@ -2520,122 +2604,139 @@ Slice::Python::CodeVisitor::writeDocstring(const OperationPtr& op, DocstringMode
...
+        hasReturnValue = true;
+        if (!comment.description.empty() || needArgs)
+        {
+            _out << nl;
+        }
+        _out << nl << "Returns";
+        _out << nl << "-------";
+        _out << nl << "Ice.Future";
+        _out << nl << "    A future object that is completed with the result of "
+             << (mode == DocAsync ? "the invocation." : "the dispatch.");
```
Comment (bernardnormier) [inline cpp/src/slice2py/PythonUtil.cpp:2667]: This does not look correct. Apparently DocSync and DocAsync are about invocations, while DocDispatch and DocAsyncDispatch are about dispatches (these names aren't consistent).

File: cpp/src/slice2py/PythonUtil.cpp (hunk 2 above)
Comment (pepone) [reply cpp/src/slice2py/PythonUtil.cpp]: Fixed

File: cpp/src/slice2py/PythonUtil.cpp (hunk 3 above)
Comment (pepone) [reply cpp/src/slice2py/PythonUtil.cpp:2667]: Here we handle returning a future from Async versions, so is either DocAsycn or DocAsyncDispatch.

DocAsync -> "the invocation"
DocAsyncDispatch "the dispatch"

looks correct to me, but I might be missing your point.

Comment (bernardnormier) [reply cpp/src/slice2py/PythonUtil.cpp:2667]: You're right, I missed the `if (mode == DocAsync || mode == DocAsyncDispatch)`.

**Review by bernardnormier — APPROVED** [review]

**Pull request merged**